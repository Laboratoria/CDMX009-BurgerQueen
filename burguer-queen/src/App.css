body {

margin: 5%;
padding: 3px 10px;

}
