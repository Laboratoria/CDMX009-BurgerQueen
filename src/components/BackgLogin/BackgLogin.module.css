.containerLogin {
    background: url("../../assets/imgs/BgLogin.png") no-repeat top right;
    height: 100vh;
    width: 100vw;
}

.iconWaiter {
    background: url("../../assets/imgs/waiter.png") no-repeat top right;
    width: 205px;
    height: 195px;
    display: inline-block;
    zoom: 1;
}

.iconHatCheff {
    background: url("../../assets/imgs/hat-chef.png");
    width: 205px;
    height: 205px;
    display: inline-block;
    zoom: 1;
}

.contentWrapper {
    width: 60%;
    display: flex;
    justify-content: center;
    margin-top: 8%;
}

.wraperSeccionBottons {
    text-align: center;
}

.buttonWaiter {
    width: 230px;
    height: 60px;
    background: #ED3367;
    color: #FFFFFF;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), inset 8px 8px 5px rgba(0, 0, 0, 0.25);
    border-radius: 8px;
    font-size: 2.6em;
    font-family: 'Amaranth', sans-serif;
}

.buttonChef {
    width: 230px;
    height: 60px;
    left: 520px;
    top: 780px;
    background: #ED3367;
    color: #FFFFFF;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), inset 8px 8px 5px rgba(0, 0, 0, 0.25);
    border-radius: 8px;
    font-size: 2.6em;
    font-family: 'Amaranth', sans-serif;
}