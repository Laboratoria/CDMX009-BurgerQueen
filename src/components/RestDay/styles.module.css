.headerWrapper {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.logoRestD {
    width: 30%;
}

.iconHeader {
    display: flex;
    flex-direction: row;    
    align-items: center;
}

.iconHeader h2 {
    padding-right: 150px;
    padding-left: 30px;
    font-family: 'Amaranth', sans-serif;
    font-size: 4em;
} 

.iconWaiterRestDay {
    background: url('../../assets/imgs/waiter.png') no-repeat;
    width: 205px;
    height: 195px;
    zoom: 1;
    display: inline-block;
}

.containerAll {
    width: 100%;
    height: auto;
    display: flex;
    flex-direction: row;
    max-height: 680px;    
}

.menuWrapper {
    flex-direction: row;
    width: 60%;
    height: auto;    
}

.firstWrapperFood {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    margin-bottom: 20%;
    margin-top: 10%;
}

.secondWrapperFood {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
     margin-bottom: 20%;
}

.accountWrapper {
    width: 40%;
}

.burgersWrapper,.add,.drinks, .additional {
    background: #FFE168;
    width: 260px;
    padding: 15px;
    height: auto;
    text-align: center;
    border-radius: 14px;
    font-size: 1.5em;
    border: 1px solid hsl(0,0%,20%);
    box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.25), inset 8px 8px 8px rgba(0, 0, 0, 0.25);
}

.routeButtons {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.routeButtons a {
        align-self: center;
}
