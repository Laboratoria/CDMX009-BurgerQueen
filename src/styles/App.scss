//Header
$general-font: 'Nunito', sans-serif;
$font-header: 'Pacifico', cursive;
$width-header : 100%;
$height-header: 85px;
$font-size-header : 40px;
$white: #FDFFFC;
$color-header: #0A2463;

body{
  margin: 0px;
  font-family: $general-font;
}


.header {
  width: $width-header;
    height: $height-header;
    background-color: $color-header;
    font-family: $font-header;
    font-size: $font-size-header;
    color: $white;
    text-align: center;
    p{
      margin-top: 0;
    
    }
}

