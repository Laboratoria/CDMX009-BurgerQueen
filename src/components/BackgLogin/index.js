import React from 'react';
import Logo from '../Logo/Logo';
import { Link } from 'react-router-dom';
import styles from './BackgLogin.module.css';

const BackgLogin = () => (
    
    <div className={styles.containerLogin}>
        <Logo nameClass={"logoLogin"} />
        <div className={styles.contentWrapper}>
            <div className={styles.wraperSeccionBottons}>
                <Link to="/waiter">
                    <span className={styles.iconWaiter}></span>
                    <button type="button" className={styles.buttonWaiter}>Mesero</button>
                </Link>
            </div>
            
            <div className={styles.wraperSeccionBottons}>
                <Link to="/kitchenregister">
                    <span className={styles.iconHatCheff}></span>
                    <button type="button" className={styles.buttonChef}>Cocina</button>
                </Link>
            </div>
        </div>
    </div>
);

export default BackgLogin;
