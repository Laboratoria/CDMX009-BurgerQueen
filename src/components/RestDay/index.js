import React from 'react';
import Logo from '../Logo/Logo';
import Select from 'react-select';
import Boton from '../Boton/Boton';
import { Link } from 'react-router-dom';
import styles from './styles.module.css';
import Products from '../Products/Products';
import ReusableTable from '../ReusableTable/index';

const RestDay = ({ client, setClient }) => {

    return (
        <div>
            <div className={styles.headerWrapper}>
                <div className={styles.logoRestD}>
                    <Logo nameClass={"logoSmallLunch"} />
                </div>
                <div className={styles.iconHeader}>
                    <h2>Comidas y Cenas</h2>
                    <span className={styles.iconWaiterRestDay}></span>
                </div>
            </div>

            <div className={styles.containerAll}>
                <div className={styles.menuWrapper}>
                    <div className={styles.firstWrapperFood}>
                        <div className={styles.burgersWrapper}>
                            <div>
                                <h3>Hamburguesas</h3>
                                <Select options={Products.burgers} />
                            </div>
                        </div>

                        <div className={styles.add}>
                            <div>
                                <h3>Adicionales</h3>
                                <Select options={Products.additionals} />
                            </div>
                        </div>
                    </div>

                    <div className={styles.secondWrapperFood}>
                        <div className={styles.drinks}>
                            <div>
                                <h3>Bebidas</h3>
                                <Select options={Products.drinks} />
                            </div>
                        </div>

                        <div className={styles.additional}>
                            <div>
                                <h3>Extras</h3>
                                <Select options={Products.additionalOptions} />
                            </div>
                        </div>
                    </div>
                </div>

                <div className={styles.accountWrapper}>
                    <div>
                        <ReusableTable client={client} setClient={setClient} />
                    </div>
                </div>
            </div>

            <div className={styles.routeButtons}>
                <Link to="waiter">
                    <Boton text={"Regresar"} allstyles={"buttonOutWaiter"} />
                </Link>

                <Link to="waiterregister">
                    <Boton text={"Enviar"} allstyles={"sendToKitchen"} />
                </Link>
            </div>

        </div>
    );
}
export default RestDay;


