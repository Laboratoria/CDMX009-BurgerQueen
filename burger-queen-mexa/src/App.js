import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Login from "./components/Login";
import Personal from "./components/Personal";
import Menu from "./components/Menu";
import Breakfast from "./components/Breakfast";
import Meal from "./components/Meal";
import Floor from "./components/Floor";

import { auth } from "./components/firebase";

function App() {
  const [firebaseUser, setFirebaseUser] = React.useState(false);

  React.useEffect(() => {
    auth.onAuthStateChanged((user) => {
      console.log(user);
      if (user) {
        setFirebaseUser(user);
      } else {
        setFirebaseUser(null);
      }
    });
  }, []);
  return firebaseUser !== false ? (
    <Router>
      <Switch>
        <Route path="/" exact>
          <Login />
        </Route>
        <Route path="/Personal">
          <Personal />
        </Route>
        <Route path="/Menu">
          <Menu />
        </Route>
        <Route path="/Breakfast">
          <Breakfast />
        </Route>
        <Route path="/Meal">
          <Meal />
        </Route>
        <Route path="/Floor">
          <Floor />
        </Route>
      </Switch>
    </Router>
  ) : (
    <p>Cargando... :D</p>
  );
}

export default App;
