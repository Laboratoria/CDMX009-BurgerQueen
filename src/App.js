import React, { useState } from 'react';
import './App.css';
import RestDay from './components/RestDay';
import Kitchen from './components/Kitchen';
import 'react-bootstrap/dist/react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import BreakFast from './components/BreakFast';
import BackgLogin from './components/BackgLogin';
import InitWaiters from './components/InitWaiter';
import { products } from '../src/utils/products.js';
import WaiterAccount from './components/WaiterAccount';
import WaiterRegister from './components/WaiterRegister';
import KitchenRegister from './components/KitchenRegister';
import { Switch, Route, HashRouter } from "react-router-dom";

function App() {
  const initialDataClient = {
    numorder: '',
    namewaiter: '',
    nameclient: '',
    numtable: '',
    numpeople: '',
    items: []
  }

  const [client, setClient] = useState(initialDataClient);

  const [order, setOrder] = useState({
    items: []
  });
  const setNumorder = (numorder) => setClient ({...client, numorder})

  const addItemToOrder = (product) => {
    let newItems = [];

    if (client.items.find((i) => i.id === product.id)) {
      newItems = client.items.map((item) => {
        if (item.id === product.id) {
          return { ...item, quantity: item.quantity + 1 };
        }
        return item;
      })
    } else {
      newItems = [...client.items, { ...product, quantity: 1 }]
    }

    setClient({
      ...client,
      items: newItems
    })
  }

  const deleteItem = (position) => {
    if (position > -1) {
      client.items.splice(position, 1);
    }

    setClient({
      ...client,
      items: client.items
    })
  }

  const calculateTotal = (items) => {
    let total = 0;

    items.forEach(item => {
      total = (item.quantity * item.price) + total
    })
    return total;
  }

  const calculateAmount = (price, quantity) => {
    return price * quantity;
  }

  return (
    <div className="App">

      <HashRouter>
        <Switch>
          <Route exact path="/">
            <BackgLogin />
          </Route>

          <Route exact path="/waiter">
            <InitWaiters 
            client={client} setClient={setClient} 
            order={order} setOrder={setOrder}
            setNumorder={setNumorder}
            />
          </Route>

          <Route exact path="/login">
            <BackgLogin />
          </Route>

          <Route exact path="/breakfast">
            <BreakFast client={client} setClient={setClient} 
            products={products} 
            addItemToOrder={addItemToOrder} 
            deleteItem={deleteItem} 
            setNumorder={setNumorder}
            />
          </Route>

          <Route exact path="/restday">
            <RestDay 
            client={client} setClient={setClient}
            />
          </Route>

          <Route exact path="/waiterregister">
            <WaiterRegister 
            client={client} setClient={setClient}
            />
          </Route>

          <Route exact path="/kitchenregister">
            <KitchenRegister 
            client={client} setClient={setClient} 
            />
          </Route>

          <Route exact path="/waiteraccount/:orderId">
            <WaiterAccount
            client={client} setClient={setClient}
            calculateTotal={calculateTotal}
            calculateAmount={calculateAmount}
            />
          </Route>

          <Route exact path="/kitchen/:orderId">
            <Kitchen 
            client={client} setClient={setClient}
            order={order} setOrder={setOrder}
            />
          </Route>
        </Switch>
      </HashRouter>
    </div>
  );
}

export default App;
