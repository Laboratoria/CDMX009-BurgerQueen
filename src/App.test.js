import React from 'react';
import { render, getAllByText } from '@testing-library/react';
import App from './App';
import RestDay from './components/RestDay'

import BackgLogin from './components/BackgLogin';
import InitWaiters from './components/InitWaiter';

/* test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/waiter/i);
  expect(linkElement).toBeInTheDocument();
}); */

/* test('renders learn react link', () => {
  const { getByRole } = render(<App />);
  expect(screen.getByRole('button')).not.toHaveAttribute('disabled')
}); */

describe('AplicatiÃ³n', () => {
  test('Represent the burger queen app', () => {
    render( < App /> );
  });
});

it('should take a snapshot', () => {
  const { asFragment } = render(<App />)  
  expect(asFragment(<App />)).toMatchSnapshot()
 });

  test('renders Mesero', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Mesero/i);
  expect(linkElement).toBeInTheDocument();
}); 

test('renders Cocina', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Cocina/i);
  expect(linkElement).toBeInTheDocument();
}); 
