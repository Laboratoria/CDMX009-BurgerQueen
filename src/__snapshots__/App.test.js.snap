// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="containerLogin"
    >
      <div
        class="logoWrapper"
      >
        <img
          alt=""
          class="logoLogin"
          src="Logo_BQ.png"
        />
      </div>
      <div
        class="contentWrapper"
      >
        <div
          class="wraperSeccionBottons"
        >
          <a
            href="#/waiter"
          >
            <span
              class="iconWaiter"
            />
            <button
              class="buttonWaiter"
              type="button"
            >
              Mesero
            </button>
          </a>
        </div>
        <div
          class="wraperSeccionBottons"
        >
          <a
            href="#/kitchenregister"
          >
            <span
              class="iconHatCheff"
            />
            <button
              class="buttonChef"
              type="button"
            >
              Cocina
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
